tools/cli/lib/installcmd.c

```diff
@@ -191,6 +191,11 @@ TDNFCliAlterCommand(
                   pContext,
                   nAlterType,
                   &pSolvedPkgInfo);
+
+    if (!pSolvedPkgInfo) {
```

Comment:
you are changing the contract of the function here by sometimes returning a null without an error being set. please make a new error code and return it, then handle it here appropriately

Reply:
It's goto  cleanup; I used goto error; because it was doing nothing but going to cleanup & pSolvePkgInfo will be NULL only when package is already installed, so it's not an error condition, we can set dwError to 0.